Index: OpenJPEG_v1_3/libopenjpeg/t1.c
===================================================================
--- OpenJPEG_v1_3.orig/libopenjpeg/t1.c
+++ OpenJPEG_v1_3/libopenjpeg/t1.c
@@ -320,9 +320,9 @@ static void t1_updateflags(flag_t *flags
 	__m64 tmp2 = *(__m64*)((void*)&flagsp[-1         ]);
 	__m64 tmp3 = *(__m64*)((void*)&flagsp[-1 + stride]);
 
-	tmp1 = _mm_or_si64(tmp1, mod[s]);
-	tmp2 = _mm_or_si64(tmp2, mod[s+2]);
-	tmp3 = _mm_or_si64(tmp3, mod[s+4]);
+	tmp1 = _mm_or_si64(tmp1, (__m64)mod[s]);
+	tmp2 = _mm_or_si64(tmp2, (__m64)mod[s+2]);
+	tmp3 = _mm_or_si64(tmp3, (__m64)mod[s+4]);
 
 	*(__m64*)((void*)&flagsp[-1 - stride]) = tmp1;
 	*(__m64*)((void*)&flagsp[-1         ]) = tmp2;
