diff -up OpenJPEG_v1_3/codec/CMakeLists.txt.libm OpenJPEG_v1_3/codec/CMakeLists.txt
--- OpenJPEG_v1_3/codec/CMakeLists.txt.libm	2007-11-27 06:38:52.000000000 -0600
+++ OpenJPEG_v1_3/codec/CMakeLists.txt	2009-06-19 09:20:59.473980422 -0500
@@ -46,10 +46,6 @@ FOREACH(exe j2k_to_image image_to_j2k)
   ADD_TEST(${exe} ${EXECUTABLE_OUTPUT_PATH}/${exe})
   # calling those exe without option will make them fail always:
   SET_TESTS_PROPERTIES(${exe} PROPERTIES WILL_FAIL TRUE)
-  # On unix you need to link to the math library:
-  IF(UNIX)
-    TARGET_LINK_LIBRARIES(${exe} m)
-  ENDIF(UNIX)
   # Install exe
   INSTALL_TARGETS(/bin/ ${exe})
 ENDFOREACH(exe)
diff -up OpenJPEG_v1_3/libopenjpeg/CMakeLists.txt.libm OpenJPEG_v1_3/libopenjpeg/CMakeLists.txt
--- OpenJPEG_v1_3/libopenjpeg/CMakeLists.txt.libm	2009-06-19 09:18:44.021229314 -0500
+++ OpenJPEG_v1_3/libopenjpeg/CMakeLists.txt	2009-06-19 09:18:44.074248015 -0500
@@ -35,6 +35,11 @@ ADD_LIBRARY(${OPENJPEG_LIBRARY_NAME} ${O
 SET_TARGET_PROPERTIES(${OPENJPEG_LIBRARY_NAME} PROPERTIES
 	${OPENJPEG_LIBRARY_PROPERTIES})
 
+IF(UNIX)
+  TARGET_LINK_LIBRARIES(${OPENJPEG_LIBRARY_NAME} m)
+  TARGET_LINK_LIBRARIES(${OPENJPEG_LIBRARY_NAME} LINK_INTERFACE_LIBRARIES)
+ENDIF(UNIX)
+
 # Install library
 INSTALL(TARGETS ${OPENJPEG_LIBRARY_NAME}
 	DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff -up OpenJPEG_v1_3/mj2/CMakeLists.txt.libm OpenJPEG_v1_3/mj2/CMakeLists.txt
--- OpenJPEG_v1_3/mj2/CMakeLists.txt.libm	2007-09-12 04:04:03.000000000 -0500
+++ OpenJPEG_v1_3/mj2/CMakeLists.txt	2009-06-19 09:18:44.074248015 -0500
@@ -10,33 +10,21 @@ ADD_EXECUTABLE(frames_to_mj2
   compat/getopt.c 
   mj2_convert.c mj2.c )
 TARGET_LINK_LIBRARIES(frames_to_mj2 ${OPJ_PREFIX}openjpeg)
-IF(UNIX)
-  TARGET_LINK_LIBRARIES(frames_to_mj2 m)
-ENDIF(UNIX)
 
 ADD_EXECUTABLE(mj2_to_frames
     mj2_to_frames.c
     compat/getopt.c mj2_convert.c mj2.c )
 TARGET_LINK_LIBRARIES(mj2_to_frames ${OPJ_PREFIX}openjpeg)
-IF(UNIX)
-  TARGET_LINK_LIBRARIES(mj2_to_frames m)
-ENDIF(UNIX)
 
 ADD_EXECUTABLE(extract_j2k_from_mj2
     extract_j2k_from_mj2.c
     mj2.c )
 TARGET_LINK_LIBRARIES(extract_j2k_from_mj2 ${OPJ_PREFIX}openjpeg)
-IF(UNIX)
-  TARGET_LINK_LIBRARIES(extract_j2k_from_mj2 m)
-ENDIF(UNIX)
 
 ADD_EXECUTABLE(wrap_j2k_in_mj2
     wrap_j2k_in_mj2.c
     mj2.c )
 TARGET_LINK_LIBRARIES(wrap_j2k_in_mj2 ${OPJ_PREFIX}openjpeg)
-IF(UNIX)
-  TARGET_LINK_LIBRARIES(wrap_j2k_in_mj2 m)
-ENDIF(UNIX)
 
 INSTALL(TARGETS frames_to_mj2 mj2_to_frames extract_j2k_from_mj2 wrap_j2k_in_mj2
 	DESTINATION bin)
