diff --git a/src/lib/openjp3d/openjp3d.h b/src/lib/openjp3d/openjp3d.h
index 0dd8aed..7b78ba6 100644
--- a/src/lib/openjp3d/openjp3d.h
+++ b/src/lib/openjp3d/openjp3d.h
@@ -71,7 +71,7 @@ defined with this macro as being exported.
 #endif /* !OPJ_STATIC || !WIN32 */
 
 #ifndef __cplusplus
-#if defined(HAVE_STDBOOL_H)
+#if 1
 /*
 The C language implementation does correctly provide the standard header
 file "stdbool.h".
